File: relationship-manager/risk-profiling-service/src/main/java/com/bank/rm/risk/dto/RiskDtos.java
package com.bank.rm.risk.dto;

import com.bank.rm.common.dto.RiskCategory;
import java.util.Map;
import java.util.UUID;

public class RiskDtos {

    public record CustomerRiskInput(
        UUID customerId, int ageMidpoint, double annualIncome, double savings,
        double investments, double investmentDiversity, int retirementHorizon,
        double equityExposure, double monthlyExpenses, int dependents,
        double riskTolerance, double employmentStability, double insuranceCoverage,
        double emergencyFundMonths, double debtToIncome, double financialLiteracy
    ) {}

    public record RiskScoringResult(
        double riskScore, RiskCategory riskCategory, Map<String, Double> featureVector,
        Map<String, Double> shapValues, String modelVersion
    ) {}

    public record RiskAssessmentRequest(
        int ageMidpoint, double annualIncome, double savings, double investments,
        double investmentDiversity, int retirementHorizon, double equityExposure,
        double monthlyExpenses, int dependents, double riskTolerance,
        double employmentStability, double insuranceCoverage,
        double emergencyFundMonths, double debtToIncome, double financialLiteracy
    ) {}

    public record RiskAssessmentResponse(
        UUID assessmentId, UUID customerId, double riskScore, RiskCategory riskCategory,
        String explanation, Map<String, Double> shapValues,
        Map<String, double[]> portfolioAllocation, String modelVersion
    ) {}

    private RiskDtos() {}
}

File: relationship-manager/risk-profiling-service/src/main/kotlin/com/bank/rm/risk/dto/RiskDtos.kt
package com.bank.rm.risk.dto

import java.util.UUID

data class RiskAssessmentRequest(
    val ageMidpoint: Int,
    val annualIncome: Double,
    val currentSavings: Double,
    val investmentTypes: List<String> = emptyList(),
    val retirementAge: Int = 60,
    val equityPercent: Double = 0.0,
    val debtPercent: Double = 0.0,
    val monthlySurplusRatio: Double = 0.0,
    val dependents: Int = 0,
    val statedRiskTolerance: Double = 5.0,
    val yearsEmployed: Int = 0,
    val hasInsurance: Boolean = false,
    val hasEmergencyFund: Boolean = false,
    val debtToIncomeRatio: Double = 0.0,
    val financialLiteracyScore: Double = 5.0
)

data class RiskAssessmentResponse(
    val assessmentId: UUID,
    val customerId: UUID,
    val riskScore: Double,
    val riskCategory: String,
    val explanation: String?,
    val topFactors: List<FactorContribution> = emptyList(),
    val portfolioAllocation: PortfolioAllocationResponse? = null,
    val modelVersion: String,
    val assessedAt: String
)

data class FactorContribution(
    val factor: String,
    val shapValue: Double
)

data class PortfolioAllocationResponse(
    val equity: Double,
    val debt: Double,
    val gold: Double,
    val cash: Double,
    val alternative: Double
)
